lib/Extractor/Candidates.cpp

```diff
@@ -860,6 +864,70 @@ std::tuple<BlockPCs, std::vector<InstMapping>> GetRelevantPCs(
   return std::make_tuple(RelevantBPCs, RelevantPCs);
 }
 
+void PrintDataflowInfo(Function &F, Instruction &I, LazyValueInfo *LVI,
+                       ScalarEvolution *SE) {
+  auto V = I.getOperand(0);
+  auto DL = F.getParent()->getDataLayout();
+  if (PrintNegAtReturn) {
+    bool Negative = isKnownNegative(V, DL);
+    if (Negative)
+      llvm::outs() << "known negative from compiler: " << "true" << "\n";
+    else
+      llvm::outs() << "known negative from compiler: " << "false" << "\n";
+  }
+  if (PrintNonNegAtReturn) {
+    bool NonNegative = isKnownNonNegative(V, DL);
+    if (NonNegative)
+      llvm::outs() << "known nonNegative from compiler: " << "true" << "\n";
+    else
+      llvm::outs() << "known nonNegative from compiler: " << "false" << "\n";
+  }
+  if (PrintKnownAtReturn) {
+    unsigned Width = DL.getTypeSizeInBits(V->getType());
+    KnownBits Known(Width);
+    computeKnownBits(V, Known, DL);
+    llvm::outs() << "knownBits from compiler: "
+                 << Inst::getKnownBitsString(Known.Zero, Known.One) << "\n";
+  }
+  if (PrintPowerTwoAtReturn) {
+    bool PowerTwo = isKnownToBeAPowerOfTwo(V, DL);
+    if (PowerTwo)
+      llvm::outs() << "known powerOfTwo from compiler: " << "true" << "\n";
+    else
+      llvm::outs() << "known powerOfTwo from compiler: " << "false" << "\n";
+  }
+  if (PrintNonZeroAtReturn) {
+    bool NonZero = isKnownNonZero(V, DL);
+    if (NonZero)
+      llvm::outs() << "known nonZero from compiler: " << "true" << "\n";
+    else
+      llvm::outs() << "known nonZero from compiler: " << "false" << "\n";
+  }
+  if (PrintSignBitsAtReturn) {
+    unsigned NumSignBits = ComputeNumSignBits(V, DL);
+    if (NumSignBits > 1)
+      llvm::outs() << "known signBits from compiler: " << NumSignBits << "\n";
```

Comment:
only say "known" for the actual knownBits analysis, it's part of the name of that analysis, but it's not part of the name of the other ones

Reply:
sure, will fix it. I used known as a verb here, but I agree it gets confusing with the actual analysis name.